2009-flu-uk.qmd

```diff
@@ -1 +1,437 @@
+
+```{r}
+data$time <- data$Days
+filter <- dust_unfilter_create(seir, data = data, time_start = 0)
+dust_likelihood_run(filter, list(alpha=par0[1],
+                                    R_0=par0[2],
+                                    h_times = hol_t,
+                                    h_values= hol_v))
+```
+
+This is the likelihood associated with the dust model written using odin. We know need to move this into the monty statistical world to be able to integrate this into our pipeline. For this use the `packer` concept that allows us to definne a one-to-one translation between the two worlds.
+
+```{r}
+packer <- monty_packer(c("alpha", "R_0", "rho", "eta"),
+                       fixed = list(h_times = hol_t, h_values= hol_v))
+```
+
+This defines two functions that allows to 'pack' (define) and 'unpack' (define) our parameters.
+
+Now building the `monty` model for the likelihood of the model is very easy:
+
+```{r}
+likelihood <- dust_likelihood_monty(filter, packer)
+```
+
+### Setting up the prior distribution
+
+We similarly set up a log_prior function. Note that we use the function to exclude impossible values such as proportion below 0 or above 1. This is to avoid to get logdensity of -infinity given that these values would have a density value of 0 (=impossible).
+
+The other ingredient we need is a prior. This we can construct with monty_dsl as before, note that the names are matching the parameters from the likelihood as it should be.
+
+```{r}
+prior <- monty_dsl({
+  alpha ~ Beta(a = 4e-4, b = 2)
+  R_0 ~ Gamma(2, 0.7)
+  rho ~ Uniform(0, 1)
+  eta ~ Exponential(mean = 1000)
+})
+```
+
+Let's inspect the our prior `monty` model
+```{r}
+prior
+```
+
+As it has been built using the monty DSL, it comes with gradient calculation, direct sampling (so no need to use one of the Monte Carlo algorithm) and accepts multiple parameters.
+
+Let's try to generate samples and check that they match our prior distribution:
+```{r}
+n_streams <- 10000
+r <- monty_rng$new(n_streams = n_streams)
+prior_samples <- matrix(monty_model_direct_sample(prior, r), nrow = n_streams)
+colnames(prior_samples) <- prior$parameters
+```
+
+```{r}
+bayesplot::mcmc_pairs(prior_samples)
+```
+
+### Setting up the posterior
+
+Our posterior is the product of the likelihood and prior, or the sum of their logs:
+
+```{r}
+posterior <- likelihood + prior
+```
+
+### Chosing the sampler
+
+Next, we define a sampler; we'll start with a random walk with a fairly arbitrary diagonal proposal matrix:
+
+```{r}
+vcv <- diag(c(5e-10,5e-5,1e-5,1))
+sampler <- monty_sampler_adaptive(vcv, initial_vcv_weight = 100)
+```
+
+We start this off, using explicit initial conditions
+
+```{r}
+#| cache: true
+init_values <- packer$pack(list(alpha=par0[1],
+                  R_0=par0[2],
+                  rho=0.1,
+                  eta=10,
+                  h_times = hol_t,
+                  h_values= hol_v))
```

Comment:
```suggestion
init_values <- packer$pack(list(alpha = par0[1],
                  R_0 = par0[2],
                  rho = 0.1,
                  eta = 10,
                  h_times = hol_t,
                  h_values= hol_v))
```

Reply:
actually, as this bit seems superfluous, delete?

Reply:
Done